class OneException(message: String) extends Exception {
  fun getMessage(): String {
    this.message
  }
}

class AnotherException(message: String) extends Exception {
  fun getMessage(): String {
    this.message
  }
}

fun main(): void {
  print_raw(
    try {
      if (true) {
        throw (OneException("true\n"))
      } else {
        throw (AnotherException("false"))
      }
    } catch {
    | OneException(message) -> message
    | e -> e.getMessage()
    }
  )
}
